### PR TITLE
CORE-527: Support `.api_only` for all blockchains.

### PR DESCRIPTION
**Pull request by EBGToo**
Requires Java changes.

### CODE REVIEW

File: Swift/BRCrypto/BRCryptoSystem.swift [hunk 1]
```diff
@@ -199,14 +199,21 @@ public final class System {
...
+    ///
+    /// It is possible that the `.api_only` mode does not work - for exmaple, the BDB is down.  In
+    /// that case it is an App issue to report and resolve the issue by: waiting out the outage;
+    /// selecting another mode if available.
+    ///
     var supportedModesMap: [String:[WalletManagerMode]] = [
-        "bitcoin-mainnet":      [.p2p_only],
-        "bitcoincash-mainnet":  [.p2p_only],
+        "bitcoin-mainnet":      [.api_only, .p2p_only],
+        "bitcoincash-mainnet":  [.api_only, .p2p_only],
```
Comment (michael-brd) [inline Swift/BRCrypto/BRCryptoSystem.swift]: Is BCH in  BDB?

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: So do we track BDB rollouts or do we let the WalletManager sync fail (API_ONLY selected, BDB produces no results or errors)?

Comment (michael-brd) [reply Swift/BRCrypto/BRCryptoSystem.swift]: I think releases lock down what they support. That way we limit potential unexpected behaviour.

Comment (EBGToo) [reply Swift/BRCrypto/BRCryptoSystem.swift]: CORE-542 created.

**Pull request merged**